File: server/internal/entities/exchange_rate_with_currency.go
package entities

type ExchangeRateWithCurrency struct {
	ID   string  `json:"id"`
	Code string  `json:"code"`
	Name string  `json:"name"`
	Base int     `json:"base"`
	Date string  `json:"date"`
	Rate float64 `json:"rate"`
}

File: app/lib/entities/exchange_rate/exchange_rate_with_currency.dart
final class ExchangeRateWithCurrency {
  const ExchangeRateWithCurrency({
    required this.id,
    required this.code,
    required this.name,
    required this.base,
    required this.date,
    required this.rate,
  });

  final String id;
  final String code;
  final String name;
  final int base;
  final String date;
  final double rate;

  factory ExchangeRateWithCurrency.fromJson(Map<String, dynamic> json) {
    return ExchangeRateWithCurrency(
      id: json["id"] as String,
      code: json["code"] as String,
      name: json["name"] as String,
      base: json["base"] as int,
      date: json["date"] as String,
      rate: json["rate"] as double,
    );
  }
}
